sync-core/src/main/java/com/cloudant/sync/datastore/DatastoreExtended.java

```diff
@@ -119,13 +121,26 @@
      *            from a remote datastore.
      * @param revisionHistory The history of the revision being inserted,
      *                        including the rev ID of {@code rev}. This list
-     *                        needs to be sorted in ascending order.
-     * @param preparedAttachments Attachments that have already been prepared, this is a
-     *                            Map of String[docId,revId], list of attachments
+     *                        needs to be sorted in ascending order
+     * @param attachments Attachments metadata and optionally data if {@code pullAttachmentsInline} true
+     * @param preparedAttachments Non-empty if {@code pullAttachmentsInline} false.
+     *                            Attachments that have already been prepared, this is a
+     *                            Map of String[docId,revId] → list of attachments
+     * @param pullAttachmentsInline
```

Comment:
The description of this parameter is detached from the parameter name - it should probably be 2 separate sentences attached to the parameter name rather than 2 paragraphs.